dash/_pages.py

```diff
@@ -358,7 +358,7 @@ def register_page(
     # Sort by order and module, then by module
     for page in sorted(
         PAGE_REGISTRY.values(),
-        key=lambda i: (str(i.get("order", i["module"])), i["module"]),
+        key=lambda i: (str(i.get("order", i["module"])).zfill(9), i["module"]),
```

Comment:
Thanks @gothicVI! This seems like it would fix a lot of common cases, feels a little arbitrary though, and IDK if anyone does this but it would also prevent using fractional orders, in case you want to squeeze a new page between two existing pages.

Also, I just had a chat with @AnnMarieW about this, if you provide a string for `order` it will interleave with `module` names - is that what we want? Or do we want everything that has a defined `order` to come before anything that doesn't? @AnnMarieW points out that if we change that it could be considered a breaking change, but I think we could argue this is also a bug and we should fix it.

My proposal then would be we order as:
- all defined numeric orders
- then all defined string orders
- with module as a secondary sort only when orders are equal.

which we could write as perhaps:
```suggestion
        key=lambda i: (
            "order" not in i,  # False (order given) sorts before True
            i["order"] if isinstance(i.get("order"), (int, float)) else float("inf"),
            str(i.get("order")),
            i["module"],
        ),
```
Whatever we end up with, let's extend [test_pages_order](https://github.com/plotly/dash/blob/dev/tests/integration/multi_page/test_pages_order.py) to lock in the fixed behavior.

Reply:
Hi @alexcjohnson, hi @AnnMarieW,

sure that would work as well.
You should be able to add it - if not I'll commit the suggestion.

How to go about the change in https://github.com/plotly/dash/blob/dev/tests/integration/multi_page/test_pages_order.py?
Do we just add more pages? - I'll suggest something :smile: